### PR TITLE
[Merged by Bors] - Basic Error Reporting Help

### PR DESCRIPTION
**Pull request by DmitryDodzin**
relevant #582

### CODE REVIEW

**Review by aviramha — COMMENTED** [review]
LGTM.
I think we need a message for the user to check the logs being sent before uploading though.

File: mirrord-layer/src/lib.rs [hunk 1]
```diff
@@ -355,9 +369,9 @@ async fn thread_loop(
         impl tokio::io::AsyncRead + tokio::io::AsyncWrite + Unpin + Send,
         ClientCodec,
     >,
-    steal: bool,
+    config: LayerConfig,
 ) {
-    let mut layer = Layer::new(codec, steal);
+    let mut layer = Layer::new(codec, config.feature.network.incoming.is_steal());
```
Comment (eyalb181) [inline mirrord-layer/src/lib.rs:374]: How did this sneak in here?

File: mirrord-layer/src/tracing_util.rs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            r#"
+mirrord full log file at: {}"#,
+            log_file.display()
+        );
+
+        let issue_link = create_github_link(log_file);
+
+        println!(
+            r#"
+mirrord encountered an error, please create an issue over at github, It would be much appreciated
```
Comment (eyalb181) [inline mirrord-layer/src/tracing_util.rs:65]:
```suggestion
mirrord encountered an error. We'd appreciate it if you could create an issue on our GitHub repository so we could address it as soon as possible.```

File: mirrord-layer/src/tracing_util.rs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+NOTE: Please check and reduct the logs in the issue for sensative information. (We do love hearing feedback on how can we reduce the amount of sensative information in the logs)
```
Comment (eyalb181) [inline mirrord-layer/src/tracing_util.rs:67]:
```suggestion
NOTE: Please redact sensitive information from the logs.
```

**Review by eyalb181 — CHANGES_REQUESTED** [review]

File: mirrord-layer/src/lib.rs (hunk 1 above)
Comment (DmitryDodzin) [reply mirrord-layer/src/lib.rs:374]: wdym?

**Review by eyalb181 — APPROVED** [review]

**Pull request closed without merging**